test/test_transforms.py

```diff
@@ -433,6 +446,30 @@ def test_batch_spectrogram(self):
         self.assertTrue(computed.shape == expected.shape, (computed.shape, expected.shape))
         self.assertTrue(torch.allclose(computed, expected))
 
+    def test_batch_melspectrogram(self):
+        waveform, sample_rate = torchaudio.load(self.test_filepath)
+
+        # Single then transform then batch
+        expected = transforms.MelSpectrogram()(waveform).repeat(3, 1, 1, 1)
```

Comment:
This is repetitive. I'm wondering if there is a way of creating a test generator that explicitly exercises this decorator. It's a function that accepts a function with some args and kwargs and then assumes that the first input is to be batchable (so it applies various reshapes to args[0] etc.)

```
def _gen_batchable(self, func, *args, **kwargs):
    self.assertEqual(func(*args, **kwargs), func(*(args[0].reshape(3, -1, -1, -1) + args[1:]), **kwargs)
```
(not sure on the reshape behavior, could also be a lambda).

Reply:
I followed the format that I did for testing jitability, and simply introduced common functions to test batching. Thoughts?

Reply:
Yeah I'm mostly just trying to iterate on top of that. Might be one of these non-problems I'm trying to fix.